### ISSUE TITLE
Uncommon launcher crash while installing mods; leaves the temporary files incomplete

### ISSUE THREAD

**Issue by drown3d** [PrismLauncher/PrismLauncher#2619]
### Operating System

Windows

### Version of Prism Launcher

Prism Launcher 8.4

### Version of Qt

Qt 6.6.3

### Description of bug

The launcher will occasionally crash when installing mods in bulk, whether the game is running or not. This will leave at least one ".jar.[random string of six letters]" file in the mods folder that has to be removed manually.

### Steps to reproduce

* Install more than one mod to any instance at once

### Suspected cause

It seems to be random, but the log is showing "Failed to count hard links for [mod directory] : 'The system cannot find the file specified.'"
[PrismLauncher-1.log](https://github.com/user-attachments/files/16118032/PrismLauncher-1.log)

### This issue is unique

- [X] I have searched the issue tracker and did not find an issue describing my bug.

**Comment by Trial97** [comment]
No the file is not corrupted, it is incomplete. Prism downloads first the mod to that file and once complete renames the file (and deletes it if you interrupt the download). The deletion of the file is not happening due to the crash. You are safe to remove that file and redownload the mod.

Is that the crash log or the log after the crash?
With what number of mods does this reproduce? Are the mods selected from both providers?
Did you change the number of concurrent downloads? if yes what is the new number(default is 6)?

**Comment by drown3d** [comment]
This is the log after the crash, as I'm not sure where crash logs are stored.
It happened recently with 4-6 mods from modrinth only
And no i didn't change that number.
Sorry if i was a bit vague or misleading, i don't have as much technical knowhow as seemingly most of github lmao

**Comment by Trial97** [comment]
I'm not able to reproduce this and there is no way for me to figure this out just from the logs.
No there is no crash log(generated by prism) per se. Maybe a core dump but I don't know where on Windows is.
No issue with the language I just wanted to make sure I did not miss it.

**Comment by TipsyTheCat** [comment]
This was happening to me until i lowered the "concurrent tasks" setting from 10 to 8. Not sure how running to many tasks at once causes a crash...

**Comment by TipsyTheCat** [comment]
should probably limit the max value for "concurrent tasks"

**Comment by Trial97** [comment]
> should probably limit the max value for "concurrent tasks"

Is limited to 99

**Comment by mirronake** [comment]
I had the exact same problem.

[PrismLauncher-1.log](https://github.com/user-attachments/files/16997041/PrismLauncher-1.log)

**Comment by Trial97** [comment]
Hi regarding the crash I prepared a portable debug build: https://media.githubusercontent.com/media/Trial97/prism_gdb/refs/heads/master/PrismLauncher-Windows-MinGW-w64-Portable-b1113f4-Debug.zip?download=true (portable build + debuger)

Download and extract that file.
Start prism using the `gdbprismlauncher.bat` file(the extension may be missing), not the normal executable; do not touch the opened command prompt(black thing with text)
Try to reproduce the crash;if the application is stuck leave it a few minutes(it takes a while to generate all the backtrace)

Once it completely closes there will be a `gdbcrash.log` file. Please share that file( is a stacktrace of the crash)

The source code for how I created that zip can be found here https://github.com/Trial97/prism_gdb/tree/master

**Comment by hugo9655** [comment]
@Trial97 i also have this issue. it happens when selecting multiple mods (~7+) and installing all at once, which causes prism to crash and half the mods to not be installed (some mods wont install properly and are empty/bugged or are incomplete with random letters at the end).

should i try to reproduce with that debug build?

**Comment by Trial97** [comment]
> should i try to reproduce with that debug build?

yes, but if you want a more recent build follow the steps mentioned in https://github.com/Trial97/prism_gdb/tree/master
(if you can reproduce it just mention if you used my link or built yourself the zip using the steps)

**Comment by AnAwesomGuy** [comment]
i also have this exact same problems, my concurrent tasks was set to 10 and concurrent downloads was 6, and when downloading around more than 7 mods it would always crash. this is the first time i encountered this bug (it started this year). i havent changed any settings, or updated or anything, it seemingly started happening out of nowhere. my log also displays the same problem as the OP.

ill try the debug stuff soon.

also, lowering the concurrent downloads/tasks fixes this completely.

**Comment by AnAwesomGuy** [comment]
@Trial97 the debug stuff doesnt seem to work, i downloaded the debug prism (https://nightly.link/PrismLauncher/PrismLauncher/workflows/trigger_builds/develop/PrismLauncher-Windows-MinGW-w64-Portable-0b6ce55-Debug.zip), `gbd.exe`, `gdbprismlauncher.bat`, `.gdbinit` and ran the bat file, but nothing happens, the command prompt instantly closes

edit: using the .zip from up there, i was able to get it working, and now i just cant seem to reproduce it. so odd...

**Comment by Trial97** [comment]
As no info was provided(from the issue author) I will close this issue.
Please provide the mentioned log before reopening it.

**Comment by hugo9655** [comment]
@Trial97 hey just so u know i wasnt able to reproduce any of the issues in the dev build. i tried multiple times in different ways but it always worked flawlessly

**Comment by ambientNexus** [comment]
Started to occur again with recent versions of prism.
